src/utils/typescript.ts

```diff
@@ -146,3 +164,87 @@ export const enableSourceMaps = (): void => {
 
     processWithTranspileSymbol[TESTPLANE_TRANSFORM_HOOK].enableSourceMaps();
 };
+
+function removeViteQueryImports(code: string, sourceFile: string): string {
+    const extname = path.extname(sourceFile);
+    const isJsxOrTsx = extname === ".jsx" || extname === ".tsx";
+
+    if (!isJsxOrTsx) {
+        return code;
+    }
+
+    let parser: RecastParser | null = null;
+
+    try {
+        // eslint-disable-next-line @typescript-eslint/no-var-requires
+        const babelParser: typeof import("@babel/parser") = require("@babel/parser");
+
+        parser = {
+            parse(source: string): ReturnType<typeof babelParser.parse> {
+                return babelParser.parse(source, {
+                    sourceType: "module",
+                    plugins: [
+                        "typescript",
+                        "jsx",
+                        "decorators-legacy",
+                        "classProperties",
+                        "nullishCoalescingOperator",
+                        "optionalChaining",
+                    ],
+                    tokens: true,
+                });
+            },
+        };
+    } catch (err) {
+        // If @babel/parser is not installed, return original code without transformation.
+        // This is not a critical issue as the user may not be using Vite-style imports with query parameters.
+        return code;
+    }
+
+    let ast: recast.types.namedTypes.File;
+
+    try {
+        ast = recast.parse(code, {
+            parser,
+            sourceFileName: sourceFile,
+            sourceRoot: path.dirname(sourceFile),
+        });
+    } catch (err) {
+        const errorMessage = (err as Error).message;
+        logger.warn(
+            `Failed to parse file ${sourceFile} for removal of Vite query imports: ${errorMessage}. ` +
+                `The file will be processed without removing query imports.`,
+        );
+        return code;
+    }
+
+    recast.visit(ast, {
+        visitImportDeclaration(nodePath) {
+            const declaration = nodePath.value as recast.types.namedTypes.ImportDeclaration;
+            const source = declaration.source.value;
+
+            if (typeof source === "string") {
+                const extname = path.extname(source);
+
+                if (extname && extname.includes("?")) {
```

Comment:
Remove only imports which has query symbol in extension name, like - `file.svg?react`